de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractOutgoingProjectNegotiation.java

```diff
@@ -117,10 +116,10 @@ public Status run(IProgressMonitor monitor) {
 
       transfer(monitor, fileLists);
 
-      User user = session.getUser(getPeer());
-      if (user == null) throw new LocalCancellationException(null, CancelOption.DO_NOT_NOTIFY_PEER);
+      if (!getRemoteUser().isInSession())
```

Comment:
The check is pointless. The user could leave the session 1 ms after you did the check.

Reply:
I am also not satisfied with the cancellation part here, but should I really change this in this PR?

Reply:
Nope